product-scenarios/1-manage-public-partner-private-apis/1.4-delete-an-existing-api/1.4.1-delete-an-existing-api/src/test/java/org/wso2/am/scenario/tests/delete/existing/api/DeleteExistingAPIsTestCases.java

```diff
@@ -0,0 +1,233 @@
+            case PROTOTYPED:
+            case PUBLISHED:
+                changeApiState(apiName, state);
+                break;
+            case BLOCKED:
+            case DEPRECATED:
+                changeApiState(apiName, APILifeCycleState.PUBLISHED);
+                changeApiState(apiName, state);
+                break;
+            case RETIRED:
+                changeApiState(apiName, APILifeCycleState.PUBLISHED);
+                changeApiState(apiName, APILifeCycleState.DEPRECATED);
+                changeApiState(apiName, state);
+                break;
+            case CREATED:
+            default:
+//                    do nothing
+        }
+    }
+
+    private void changeApiState(String apiName, APILifeCycleState state) throws Exception {
+        APILifeCycleStateRequest updateRequest;
+        switch (state) {
+            case PROTOTYPED:
+                updateRequest = new APILifeCycleStateRequest(apiName, API_CREATOR_PUBLISHER_USERNAME,
+                        APILifeCycleState.PROTOTYPED);
+                verifyApiStatusChange(apiPublisher.changeAPILifeCycleStatus(updateRequest), state.toString());
+                break;
+            case PUBLISHED:
+                updateRequest = new APILifeCycleStateRequest(apiName, API_CREATOR_PUBLISHER_USERNAME,
+                        APILifeCycleState.PUBLISHED);
+                verifyApiStatusChange(apiPublisher.changeAPILifeCycleStatus(updateRequest), state.toString());
+                break;
+            case BLOCKED:
+                updateRequest = new APILifeCycleStateRequest(apiName, API_CREATOR_PUBLISHER_USERNAME,
+                        APILifeCycleState.BLOCKED);
+                verifyApiStatusChange(apiPublisher.changeAPILifeCycleStatus(updateRequest), state.toString());
+                break;
+            case DEPRECATED:
+                updateRequest = new APILifeCycleStateRequest(apiName, API_CREATOR_PUBLISHER_USERNAME,
+                        APILifeCycleState.DEPRECATED);
+                verifyApiStatusChange(apiPublisher.changeAPILifeCycleStatus(updateRequest), state.toString());
+                break;
+            case RETIRED:
+                updateRequest = new APILifeCycleStateRequest(apiName, API_CREATOR_PUBLISHER_USERNAME,
+                        APILifeCycleState.RETIRED);
+                verifyApiStatusChange(apiPublisher.changeAPILifeCycleStatus(updateRequest), state.toString());
+                break;
+            case CREATED:
+                default:
+//                    do nothing
+        }
+    }
+
+    private void deleteAPI(String apiName) throws Exception {
+        verifyResponse(apiPublisher.deleteAPI(apiName, API_VERSION, API_CREATOR_PUBLISHER_USERNAME));
+//        verify API not available in publisher
+        HttpResponse response = apiPublisher.getAPI(apiName, API_CREATOR_PUBLISHER_USERNAME, API_VERSION);
+        log.info("API delete response code for API \'" + apiName + "\' : " + response.getResponseCode());
```

Comment:
Any reason to log here?

Reply:
Additional logs were added to help debug in case the tests fails in testgrid environment.